extension/intents/tabs/tabs.js

```diff
@@ -371,3 +371,179 @@ intentRunner.registerIntent({
     await browser.tabs.move(tabsIds, { windowId: newWindow.id, index: 0 });
   },
 });
+
+async function collectTabsAround(center, tabs) {
+  if (tabs === undefined) {
+    return;
+  }
+
+  const tabsLeft = tabs.filter(tab => {
+    return tab.index < center.index;
+  });
+
+  const tabsRight = tabs.filter(tab => {
+    return tab.index > center.index;
+  });
+
+  for (let i = 0; i < tabsLeft.length; i++) {
+    await browser.tabs.move(tabsLeft[tabsLeft.length - i - 1].id, {
+      index: center.index - i - 1,
+    });
+  }
+
+  for (let i = 0; i < tabsRight.length; i++) {
+    await browser.tabs.move(tabsRight[i].id, { index: center.index + i + 1 });
+  }
+}
+
+async function collectRightTabs(tabs) {
+  if (tabs === undefined || tabs.length === 0) {
+    return;
+  }
+
+  const leftestTab = tabs[0];
+  const tabsRight = tabs.slice(1);
+
+  for (let i = 0; i < tabsRight.length; i++) {
+    await browser.tabs.move(tabsRight[i].id, {
+      index: leftestTab.index + i + 1,
+    });
+  }
+}
+
+function groupTabs(tabs, keyGetter) {
+  if (tabs === undefined) {
+    return {};
+  }
+
+  const groupedTabs = {};
+
+  for (let i = 0; i < tabs.length; i++) {
+    const key = keyGetter(tabs[i]);
+
+    if (groupedTabs[key] === undefined) {
+      groupedTabs[key] = [tabs[i]];
+    } else {
+      groupedTabs[key].push(tabs[i]);
+    }
+  }
+
+  return groupedTabs;
+}
+
+async function getMatchingTabs(options) {
+  const tabQuery = {
+    pinned: false,
+    windowType: "normal",
+  };
+
+  if (options.currentWindow === "true") {
+    tabQuery.currentWindow = true;
+  }
+
+  if (options.query === undefined && options.activeTab !== undefined) {
+    const activeTab = options.activeTab;
+    const activeHostname = new URL(activeTab.url).hostname;
+    tabQuery.url = "*://*." + activeHostname + "/*";
+  }
+
+  let matchingTabs = await browser.tabs.query(tabQuery);
+
+  if (options.query !== undefined) {
+    matchingTabs = matchingTabs.filter(tab => {
+      return new URL(tab.url).origin.includes(options.query);
+    });
+  }
+
+  if (options.sort_by_index === true) {
+    matchingTabs.sort((a, b) => {
+      return a.index > b.index;
+    });
+  }
+
+  return matchingTabs;
+}
+
+intentRunner.registerIntent({
+  name: "tabs.collectMentionedTabs",
+  async run(context) {
+    const activeTab = await context.activeTab();
+    const currentWindow = await browser.windows.getCurrent();
+    const allWindows = await browser.windows.getAll({
+      windowTypes: ["normal"],
+    });
+
+    const matchingTabs = await getMatchingTabs({
+      query: context.slots.query,
+      activeTab,
+      sort_by_index: true,
+      currentWindow: context.parameters.currentWindow,
```

Comment:
Generally I would try to convert the string as soon as possible, so instead of testing `currentWindow === "true"` in the other function, I would do `currentWindow: context.parameters.currentWindow === "true"` right here